### DISCUSSION TITLE
What `gl::GetUniformLocation` excepts (rust)

### DISCUSSION THREAD

**Discussion by stalker57241** [Dav1dde/glad#430] [Q&A] [answered]
I'm writing app with rust and opengl, done much work with matrices, but it isn't works. After some tests, I found, that location is -1. but shader's code contains used name. It isn't work when `name` is `&str` and used as `name.as_ptr().cast()` for casting to `*const i8` type.

Where is readable docs? May anyone tell me?

**Comment by Dav1dde** [comment, marked as answer]
Strings without length in the API like `glGetUniformLocation` need to be 0-terminated, in Rust you could/should use `CString`

**Reply by stalker57241** [reply]
thanks
